src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java

```diff
@@ -118,24 +118,38 @@ public static String upsertCredential(Secret secret) throws IOException {
 
     private static String upsertCredential(Secret secret, String namespace, String secretName) throws IOException {
         String id = null;
+        String originalId = null;
         String customSecretName = getCustomName(secret);
         if (secret != null) {
             Credentials creds = secretToCredentials(secret);
             if (creds == null)
                 return null;
+            // checking with updated secret name if custom name is not null
             id = secretName(namespace, secretName, customSecretName);
+            originalId = secretName(namespace, secretName, null);
             Credentials existingCreds = lookupCredentials(id);
+            Credentials existingOriginalCreds = lookupCredentials(originalId);
+
             final SecurityContext previousContext = ACL.impersonate(ACL.SYSTEM);
             try {
                 CredentialsStore s = CredentialsProvider.lookupStores(Jenkins.getActiveInstance()).iterator().next();
-                if (existingCreds != null) {
+                if (!originalId.equals(id)) {
```

Comment:
Are you sure that this test is relevant?
Because, we are generating 2 credentials names using the same method, and then we compare the two generated string to which we applied almost the same transformation.
Don't we need just to check if customSecretName != null && !customSecretName.equals(generatedCredentialsName) ?

shouldn't customSecretName be named customCredentialName ?

Reply:
Here we'll be checking if there is another credential (dupe) in the credentials store and if there is delete it and creaet a new one with the name provided by the user based on the annotation from the secret.name annotation.